ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/util/InclusionListUtil.java

```diff
@@ -107,41 +106,34 @@ public boolean hasCorrectCommitteeRoot(
     return committeeRoot.equals(inclusionCommitteeRoot);
   }
 
-  // TODO EIP7805 this IntList to SszList conversion to get the HTR could be improved
   public Bytes32 getInclusionListCommitteeRoot(final BeaconState state, final UInt64 slot) {
-    final IntList inclusionListCommittee =
-        beaconStateAccessors.getInclusionListCommittee(state, slot);
-    final int committeeSize = inclusionListCommittee.size();
-    final SszUInt64[] inclusionListCommitteeConverted =
-        inclusionListCommittee
-            .intStream()
-            .mapToObj(index -> SszUInt64.of(UInt64.valueOf(index)))
-            .toArray(SszUInt64[]::new);
-    final SszList<SszUInt64> inclusionCommitteeSszList =
-        SszListSchema.create(SszPrimitiveSchemas.UINT64_SCHEMA, committeeSize)
-            .of(inclusionListCommitteeConverted);
-    final Bytes32 inclusionCommitteeRoot = inclusionCommitteeSszList.hashTreeRoot();
-    return inclusionCommitteeRoot;
+    final List<UInt64> inclusionListCommittee =
+        getInclusionListCommittee(state, slot).intStream().mapToObj(UInt64::valueOf).toList();
+    final SszVector<SszUInt64> sszInclusionListCommittee =
+        schemaDefinitionsEip7805
+            .getInclusionListCommitteeRootSchema()
+            .createFromElements(inclusionListCommittee.stream().map(SszUInt64::of).toList());
```

Comment:
nit: If you create and return `SszPrimitiveVectorSchema` them you may shorten this a bit (and optimize a bit): 
```
        schemaDefinitionsEip7805
            .getInclusionListCommitteeRootSchema()
            .of(inclusionListCommittee);
```

Reply:
As you suggested: https://github.com/Consensys/teku/pull/9384
Thanks!